modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/LegacyValueAdapter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.value;
+
+
+import org.mule.sdk.api.values.Value;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Adapter of the legacy {@link org.mule.runtime.api.value.Value} to {@link org.mule.sdk.api.values.Value}
+ *
+ * @since 4.4.0
+ */
+public class LegacyValueAdapter implements Value {
+
+  private final org.mule.runtime.api.value.Value value;
+
+  public LegacyValueAdapter(org.mule.runtime.api.value.Value value) {
+    this.value = value;
+  }
+
+  @Override
+  public String getId() {
+    return value.getId();
+  }
+
+  @Override
+  public String getDisplayName() {
+    return value.getDisplayName();
+  }
+
+  @Override
+  public Set<Value> getChilds() {
+    Set<Value> values = new HashSet<>();
+    value.getChilds().forEach(v -> values.add(new LegacyValueAdapter(v)));
```

Comment:
not a big deal since this is design time exclusive, but this would be more efficient as value.getChilds().stream().map(LegacyValueAdapter::new).collect(LinkedHashSet::new)